src/service/viewer-impl.js

```diff
@@ -445,7 +452,40 @@ export class Viewer {
    * @return {!VisibilityState}
    */
   getVisibilityState() {
-    return this.visibilityState_;
+    // The viewer has informed us that the doc is no longer active, meaning
+    // the viewer closed or it swiped to another doc.
+    if (this.visibilityState_ === VisibilityState.UNLOADED ||
+        this.visibilityState_ === VisibilityState.HIDDEN) {
```

Comment:
Can we just return values as is? I don't understand why we should translate two values into one here. The `visibilityState_` and `VisibilityState` enum should be specifically the "visibility state of AMP document in different viewer conditions".